core/server/master/src/main/java/alluxio/master/meta/checkconf/ServerConfigurationChecker.java

```diff
@@ -107,6 +109,10 @@ public synchronized void regenerateReport() {
    * @return the configuration checker report
    */
   public synchronized ConfigCheckReport getConfigCheckReport() {
+    if (mConfigDirty == true) {
+      mConfigDirty = false;
+      regenerateReport();
```

Comment:
Might not matter but should we set the variable after `regenerateReport`?

Reply:
we need to set it before so that if the configurations are updated during report generation, it will be reflected in the next call to `getConfigCheckReport`